runtime/doc/lua.txt

```diff
@@ -2817,4 +2914,367 @@ range({spec})                                            *vim.version.range()*
+     -- 2
+     it:next()
+     -- 3
+<
+
+    Return: ~
+        any
+
+Iter:nextback({self})                                        *Iter:nextback()*
+    Return the next value from the end of the iterator.
+
+    Only supported for iterators on tables.
+
+    Example: >lua
+
+     local it = vim.iter({1, 2, 3, 4})
+     it:nextback()
+     -- 4
+     it:nextback()
+     -- 3
+<
+
+    Return: ~
+        any
+
+Iter:nth({self}, {n})                                             *Iter:nth()*
+    Return the nth value in the iterator.
+
+    This function advances the iterator.
+
+    Example: >lua
+
+     local it = vim.iter({ 3, 6, 9, 12 })
+     it:nth(2)
+     -- 6
+     it:nth(2)
+     -- 12
+<
+
+    Parameters: ~
+      • {n}  (number) The index of the value to return.
+
+    Return: ~
+        any
+
+Iter:nthback({self}, {n})                                     *Iter:nthback()*
+    Return the nth value from the end of the iterator.
+
+    This function advances the iterator.
+
+    Only supported for iterators on tables.
+
+    Example: >lua
+
+     local it = vim.iter({ 3, 6, 9, 12 })
+     it:nthback(2)
+     -- 9
+     it:nthback(2)
+     -- 3
+<
+
+    Parameters: ~
+      • {n}  (number) The index of the value to return.
+
+    Return: ~
+        any
+
+Iter:peek({self})                                                *Iter:peek()*
+    Peek at the next value in the iterator without consuming it.
+
+    Only iterators on tables can be peeked.
+
+    Example: >lua
+
+     local it = vim.iter({ 3, 6, 9, 12 })
+     it:peek()
+     -- 3
+     it:peek()
+     -- 3
+     it:next()
+     -- 3
+<
+
+    Return: ~
+        any
+
+Iter:peekback({self})                                        *Iter:peekback()*
+    Return the next value from the end of the iterator without consuming it.
+
+    Only supported for iterators on tables.
+
+    Example: >lua
+
+     local it = vim.iter({1, 2, 3, 4})
+     it:peekback()
+     -- 4
+     it:peekback()
+     -- 4
+     it:nextback()
+     -- 4
+<
+
+    Return: ~
+        any
+
+Iter:rev({self})                                                  *Iter:rev()*
+    Reverse an iterator.
+
+    Only iterators on tables can be reversed.
+
+    Example: >lua
+
+     local it = vim.iter({ 3, 6, 9, 12 }):rev()
+     it:collect()
+     -- { 12, 9, 6, 3 }
+<
+
+    Return: ~
+        Iter
+
+Iter:rfind({self}, {f})                                         *Iter:rfind()*
+    Find the first value in the iterator that satisfies the given predicate,
+    starting from the end.
+
+    Advances the iterator. Returns nil and drains the iterator if no value is
+    found.
+
+    Only supported for iterators on tables.
+
+    Examples: >lua
+
+     local it = vim.iter({ 1, 2, 3, 2, 1 }):enumerate()
+     it:rfind(1)
+     -- 5   1
+     it:rfind(1)
+     -- 1   1
+<
+
+    Return: ~
+        any
+
+    See also: ~
+      • Iter.find
+
+Iter:skip({self}, {n})                                           *Iter:skip()*
```

Comment:
`skipback` is useful for creating a slice.

```lua
-- Remove the first 6 elements and the last 3 elements
vim.iter(t):skip(6):skipback(3)
```

This could replace `vim.list_slice`.

Reply:
Also added `vim.iter(t):slice()`.